### PR TITLE
Handle relative urls in Location redirect header

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/core/Azure.Core/src/Pipeline/Internal/RedirectPolicy.cs [hunk 1]
```diff
@@ -168,5 +173,77 @@ public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePol
...
+                    rawBytes[i] = (byte)input[i];
+                }
+                try
+                {
+                    // We don't want '?' replacement characters, just fail.
+                    System.Text.Encoding decoder = System.Text.Encoding.GetEncoding("utf-8", System.Text.EncoderFallback.ExceptionFallback,
+                        System.Text.DecoderFallback.ExceptionFallback);
+                    return decoder.GetString(rawBytes, 0, rawBytes.Length);
+                }
+                catch (ArgumentException) { } // Not actually Utf-8
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/src/Pipeline/Internal/RedirectPolicy.cs:244]: Are you worried about swallowing this exception?

**Review by annelo-msft — APPROVED** [review]

File: sdk/core/Azure.Core/src/Pipeline/Internal/RedirectPolicy.cs (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/Internal/RedirectPolicy.cs:244]: Nope, we expect the `GetString` to throw if it fails to decode.

**Pull request merged**